lib/build.js

```diff
@@ -205,7 +217,7 @@ function build(params) {
                 {
                   Effect: 'Allow',
                   Action: ['iam:ListInstanceProfiles'],
-                  Resource: params.InstanceProfile
+                  Resource: params.Properties.SpotFleetRequestConfigData.LaunchSpecifications[0].IamInstanceProfile
```

Comment:
If you wanted to be 100% future-proof, instead of just picking the first profile, you would specify them all

```js
Resource: params.Properties.SpotFleetRequestConfigData.LaunchSpecifications
  .map((spec) => spec.IamInstanceProfle);
```

While our fleets don't do this, it is feasible that you could design a fleet that does.